apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.IntegrationTests/RestoreBucketTest.cs

```diff
@@ -0,0 +1,44 @@
+using System;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Google.Cloud.Storage.V1.IntegrationTests
+{
+    [Collection(nameof(StorageFixture))]
+    public class RestoreBucketTest
+    {
+        private readonly StorageFixture _fixture;
+
+        public RestoreBucketTest(StorageFixture fixture)
+        {
+            _fixture = fixture;
+        }
+
+        [Fact]
+        public async Task RestoreSoftDeletedBucket()
+        {
+
+            // We get bucket object using get bucket async method for soft delete bucket before deleting it. 
+            var bucket = await _fixture.Client.GetBucketAsync(_fixture.SoftDeleteBucketThree, new GetBucketOptions { SoftDeleted = false });
+            // We delete soft delete bucket.
+            try
```

Comment:
Once you create the bucket on the test, you will know for a fact that it will be empty.